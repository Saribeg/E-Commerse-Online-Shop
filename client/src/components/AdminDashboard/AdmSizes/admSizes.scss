.admin-size {
  &-new {
    border: 1px dashed #888888;
    padding: 30px;
    margin-bottom: 30px;
  }

  &-title {
    font-size: 20px;
    margin-bottom: 20px;
  }

  &-label {
    font-size: 16px;
    margin-right: 10px;
  }

  &-input {
    padding: 5px;
    outline: 1px solid #888888;
    border: none;

    &:focus {
      outline: 2px solid #888888;
    }
  }

  &-save {
    padding: 5px;
    background-color: #7ecb20;
    outline: 1px solid #888888;
    border: none;
    cursor: pointer;
    transition: all 0.4s ease;

    &-new {
      padding: 5px 30px;
    }

    &:focus {
      outline: 2px solid #888888;
    }

    &:hover {
      outline: 2px solid #888888;
      background-color: #1faa00;
    }
  }

  &-minipreloader {
    display: inline-block;
    height: 20px;
    width: 20px;
    border: 3px solid #d8d8d8;
    border-right-color: transparent;
    border-radius: 50%;
    animation: rotate 0.8s infinite linear;
  }

  &-result {
    margin-left: 30px;
    color: #ff6008;
  }

  &-table {
    display: flex;
  }

  &-row {
    display: flex;
    flex-direction: column;
    border: 1px solid #888888;
  }

  &-heading {
    // padding: 20px;
    height: 50px;
    line-height: 50px;
    text-align: center;
    border-bottom: 1px solid #888888;
  }

  &-item {
    padding: 20px;

    &-multi {
      padding: 15px;
    }

    .admin-size-minipreloader {
      margin-left: 10px;
    }
  }

  &-cell {
    display: flex;
    align-items: center;
  }

  &-msg {
    display: block;
    font-size: 13px;
    color: #ff6008;
    margin-top: 5px;
  }
}

@keyframes rotate {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.error-msg {
  color: red;
  font-size: 16px;
  padding: 15px;
}
