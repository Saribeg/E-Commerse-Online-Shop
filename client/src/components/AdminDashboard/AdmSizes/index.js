import React, { Component } from "react";
import { connect } from "react-redux";

import Preloader from "../../Preloader";

import {
  getNewSizeName,
  saveNewSizeInDb,
  getExistingSizes,
  getUpdatedSize,
  updateSizeInDb,
  updateSizesInAllProducts
} from "../../../actions/adminDashboard/admSizeActions";

import "./admSizes.scss";

class AdmSizes extends Component {
  componentDidMount() {
    this.props.getExistingSizes();
  }

  render() {
    const {
      //state
      newSizeName,
      isNewSizeFetching,
      checkNewSizeResult,
      isExistingSizesFetching,
      existingSizes,
      updatingSizes,
      sizeUpdateResults,
      isSizeUpdating,
      preUpdateValue,
      updatedSizeValue,
      errorMsg,
      fetchExistingSizesMessage,

      // actions
      getNewSizeName,
      saveNewSizeInDb,
      getUpdatedSize,
      updateSizeInDb,
      getExistingSizes,
      updateSizesInAllProducts
    } = this.props;

    let existingSizeValues = existingSizes.map(size => {
      return (
        <li className="admin-size-item" key={size._id}>
          {size.value}
        </li>
      );
    });

    let updatingSizeInputs = updatingSizes.map(size => {
      return (
        <li className="admin-size-item admin-size-item-multi" key={size._id}>
          <div className="admin-size-cell">
            <input
              type="text"
              className="admin-size-input"
              value={size.value}
              onChange={e =>
                getUpdatedSize(updatingSizes, size._id, e.target.value)
              }
            />
            <input
              type="button"
              className="admin-size-save"
              value="Save changes"
              onClick={() =>
                updateSizeInDb(
                  size._id,
                  size.value,
                  existingSizes,
                  getExistingSizes,
                  updateSizesInAllProducts
                )
              }
            />
            {preUpdateValue === size.value && isSizeUpdating ? (
              <div className="admin-size-minipreloader" />
            ) : null}
          </div>
          {updatedSizeValue === size.value && sizeUpdateResults ? (
            <span className="admin-size-msg">{sizeUpdateResults}</span>
          ) : null}
        </li>
      );
    });

    let categoriesInSizes = existingSizes.map(size => {
      let multiArr = size.categories.map(cat => {
        let sizeCat = cat.split("-");
        return (
          sizeCat[sizeCat.length - 1].charAt(0).toUpperCase() +
          sizeCat[sizeCat.length - 1].slice(1)
        );
      });

      let uniqueCats = [...new Set(multiArr)];

      let uniqueCatsString = uniqueCats.join(", ");

      return (
        <li key={size._id} className="admin-size-item">
          {uniqueCatsString}
        </li>
      );
    });

    return (
      <div className="admin-size">
        <p className="admin-size-title"> Add New Size</p>
        <div className="admin-size-new">
          <div className="admin-size-add">
            <label htmlFor="add-size" className="admin-size-label">
              Fill size value:{" "}
            </label>
            <input
              type="text"
              className="admin-size-input"
              id="add-size"
              value={newSizeName}
              onChange={e => getNewSizeName(e.target.value)}
            />
            <input
              type="button"
              className="admin-size-save admin-size-save-new"
              value="Save"
              onClick={() => saveNewSizeInDb(newSizeName, getExistingSizes)}
            />
            <span className="admin-size-result">
              {isNewSizeFetching ? (
                <div className="admin-size-minipreloader" />
              ) : (
                checkNewSizeResult
              )}
            </span>
          </div>
        </div>
        <p className="admin-size-title"> See / Update Existing Sizes</p>

        {fetchExistingSizesMessage ? (
          <span className="admin-size-error">{fetchExistingSizesMessage}</span>
        ) : null}

        {errorMsg ? <p className="error-msg">{errorMsg}</p> : null}
        <div className="admin-size-existing">
          {isExistingSizesFetching ? (
            <Preloader />
          ) : (
            <div className="admin-size-table">
              <div className="admin-size-row">
                <div className="admin-size-heading">Current size</div>
                <div className="admin-size-content">
                  <ul className="admin-size-list">{existingSizeValues}</ul>
                </div>
              </div>
              <div className="admin-size-row">
                <div className="admin-size-heading">New size (update)</div>
                <div className="admin-size-content">
                  <ul className="admin-size-list">{updatingSizeInputs}</ul>
                </div>
              </div>
              <div className="admin-size-row">
                <div className="admin-size-heading">
                  Categories sizes are in
                </div>
                <div className="admin-size-content">
                  <ul className="admin-size-list">{categoriesInSizes}</ul>
                </div>
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    newSizeName: state.admSizes.newSizeName,
    isNewSizeFetching: state.admSizes.isNewSizeFetching,
    checkNewSizeResult: state.admSizes.checkNewSizeResult,
    isExistingSizesFetching: state.admSizes.isExistingSizesFetching,
    existingSizes: state.admSizes.existingSizes,
    updatingSizes: state.admSizes.updatingSizes,
    sizeUpdateResults: state.admSizes.sizeUpdateResults,
    isSizeUpdating: state.admSizes.isSizeUpdating,
    preUpdateValue: state.admSizes.preUpdateValue,
    updatedSizeValue: state.admSizes.updatedSizeValue,
    errorMsg: state.admSizes.errorMsg,
    fetchExistingSizesMessage: state.admSizes.fetchExistingSizesMessage

  };
};

export default connect(
  mapStateToProps,
  {
    getNewSizeName,
    saveNewSizeInDb,
    getExistingSizes,
    getUpdatedSize,
    updateSizeInDb,
    updateSizesInAllProducts
  }
)(AdmSizes);
