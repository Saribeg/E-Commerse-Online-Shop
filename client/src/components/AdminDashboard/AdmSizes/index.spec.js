import React from "react";
import AdmSizes from "./index";

let isExistingSizesFetching = false;
let existingSizes = [
  {
    prices: {
      min: 63,
      max: 209.99
    },
    categories: ["women-shoes"],
    colors: [
      "black",
      "gray",
      "blue",
      "rose",
      "dark beige",
      "white",
      "red",
      "navy"
    ],
    _id: "5c73fd305adcc004e054d124",
    value: "36",
    __v: 0
  },
  {
    prices: {
      min: 88.3,
      max: 209.99
    },
    categories: ["men-shoes", "women-shoes"],
    colors: ["black", "brown", "dark beige", "rose"],
    _id: "5c73fd405adcc004e054d128",
    value: "40",
    __v: 0
  },
  {
    prices: {
      min: 122.32,
      max: 122.32
    },
    categories: ["men-shoes"],
    colors: ["black"],
    _id: "5c73fd455adcc004e054d129",
    value: "41",
    __v: 0
  },
  {
    prices: {
      min: 122.32,
      max: 122.32
    },
    categories: ["men-shoes"],
    colors: ["black"],
    _id: "5c73fd495adcc004e054d12a",
    value: "42",
    __v: 0
  },
  {
    prices: {
      min: 122.32,
      max: 122.32
    },
    categories: ["men-shoes"],
    colors: ["black"],
    _id: "5c73fd4c5adcc004e054d12b",
    value: "43",
    __v: 0
  },
  {
    prices: {
      min: 55,
      max: 55
    },
    categories: ["men-accessories-belts"],
    colors: ["gray", "burgundy"],
    _id: "5c7d1efa558cad33d8f43f4e",
    value: "76 - 81 CM",
    __v: 0
  },
  {
    prices: {
      min: 55,
      max: 55
    },
    categories: ["men-accessories-belts"],
    colors: ["gray", "burgundy", "brown", "navy blue"],
    _id: "5c7d1f0f558cad33d8f43f4f",
    value: "81 - 86 CM",
    __v: 0
  },
  {
    prices: {
      min: 55,
      max: 55
    },
    categories: ["men-accessories-belts"],
    colors: ["gray", "burgundy", "brown", "navy blue"],
    _id: "5c7d1f17558cad33d8f43f50",
    value: "86 - 91 CM",
    __v: 0
  },
  {
    prices: {
      min: 20,
      max: 655.55
    },
    categories: [
      "women-clothing-dresses",
      "women-accessories-hats",
      "men-clothing-suits",
      "men-clothing-pants",
      "women-clothing-jackets",
      "women-clothing-pants",
      "women-clothing-tops",
      "men-clothing-shirts",
      "men-clothing-sport",
      "women-clothing-hoodies"
    ],
    colors: [
      "black",
      "white",
      "dark beige",
      "blue",
      "yellow",
      "blue black",
      "gray",
      "dark gray",
      "navy blue",
      "navy",
      "burgundy",
      "light brown",
      "beige",
      "green",
      "red",
      "rose",
      "light blue",
      "orange",
      "wine"
    ],
    _id: "5c70205129b47b30640a8f73",
    value: "m",
    sizeDesc: "medium",
    __v: 0
  },
  {
    prices: {
      min: 30,
      max: 575
    },
    categories: [
      "women-clothing-dresses",
      "women-clothing-pants",
      "men-clothing-suits",
      "men-clothing-pants",
      "women-clothing-jackets",
      "women-clothing-tops",
      "women-accessories-hats",
      "men-clothing-shirts",
      "men-clothing-sport",
      "women-clothing-hoodies"
    ],
    colors: [
      "black",
      "blue",
      "light blue",
      "rose",
      "yellow",
      "gray",
      "navy blue",
      "dark gray",
      "wine",
      "white",
      "navy",
      "dark beige",
      "burgundy",
      "light brown",
      "blue black",
      "beige",
      "green",
      "red",
      "orange"
    ],
    _id: "5c700f151c7b2913f88b834c",
    value: "xs",
    sizeDesc: "extra small",
    __v: 0
  },
  {
    prices: {
      min: 25.95,
      max: 159.95
    },
    categories: ["men-accessories-belts", "women-accessories-belts"],
    colors: [
      "burgundy",
      "brown",
      "dark beige",
      "black",
      "red",
      "beige",
      "white",
      "rose"
    ],
    _id: "5c7d1f1f558cad33d8f43f51",
    value: "91 - 96 CM",
    __v: 0
  },
  {
    prices: {
      min: 63,
      max: 209.99
    },
    categories: ["women-shoes"],
    colors: [
      "black",
      "gray",
      "dark beige",
      "blue",
      "rose",
      "wine",
      "white",
      "red",
      "navy",
      "navy blue"
    ],
    _id: "5c73fd345adcc004e054d125",
    value: "37",
    __v: 0
  },
  {
    prices: {
      min: 63,
      max: 209.99
    },
    categories: ["women-shoes", "men-shoes"],
    colors: [
      "black",
      "gray",
      "brown",
      "dark beige",
      "wine",
      "white",
      "red",
      "navy",
      "navy blue"
    ],
    _id: "5c73fd385adcc004e054d126",
    value: "38",
    __v: 0
  },
  {
    prices: {
      min: 63,
      max: 209.99
    },
    categories: ["women-shoes", "men-shoes"],
    colors: [
      "gray",
      "black",
      "brown",
      "dark beige",
      "blue",
      "rose",
      "wine",
      "white",
      "red",
      "navy",
      "navy blue"
    ],
    _id: "5c73fd3c5adcc004e054d127",
    value: "39",
    __v: 0
  },
  {
    prices: {
      min: 399.99,
      max: 450
    },
    categories: ["men-accessories-clocks"],
    colors: ["brown", "black"],
    _id: "5c9923bf4f16a51810eb119f",
    value: "Standard",
    __v: 0
  },
  {
    prices: {
      min: 400,
      max: 450
    },
    categories: ["men-accessories-clocks"],
    colors: ["brown", "black"],
    _id: "5c9923db4f16a51810eb11a0",
    value: "Small",
    __v: 0
  },
  {
    prices: {
      min: 399.99,
      max: 450
    },
    categories: ["men-accessories-clocks"],
    colors: ["brown"],
    _id: "5c9923e34f16a51810eb11a1",
    value: "Midsize",
    __v: 0
  },
  {
    prices: {
      min: 400,
      max: 450
    },
    categories: ["men-accessories-clocks"],
    colors: ["brown", "black"],
    _id: "5c9923ec4f16a51810eb11a2",
    value: "XL/Oversize",
    __v: 0
  },
  {
    prices: {
      min: 23,
      max: 70
    },
    categories: [
      "women-accessories-wallets",
      "men-accessories-wallets",
      "kids-girls-toys"
    ],
    colors: [
      "white",
      "black",
      "brown",
      "burgundy",
      "blue",
      "yellow",
      "orange",
      "green",
      "beige",
      "wine",
      "rose",
      "dark beige",
      "red",
      "navy blue",
      "dark gray",
      "violet"
    ],
    _id: "5c9924354f16a51810eb11a3",
    value: "standart",
    __v: 0
  },
  {
    prices: {
      min: 20,
      max: 655.55
    },
    categories: [
      "women-clothing-dresses",
      "women-clothing-pants",
      "men-clothing-suits",
      "men-clothing-pants",
      "women-clothing-jackets",
      "women-clothing-tops",
      "women-accessories-hats",
      "men-clothing-shirts",
      "men-clothing-sport",
      "women-clothing-hoodies"
    ],
    colors: [
      "black",
      "blue",
      "light blue",
      "dark beige",
      "rose",
      "blue black",
      "gray",
      "navy blue",
      "dark gray",
      "wine",
      "white",
      "navy",
      "orange",
      "light brown",
      "beige",
      "green",
      "red",
      "burgundy",
      "yellow"
    ],
    _id: "5c70206129b47b30640a8f74",
    value: "l",
    sizeDesc: "large",
    __v: 0
  },
  {
    prices: {
      min: 20,
      max: 655.55
    },
    categories: [
      "women-clothing-dresses",
      "women-clothing-pants",
      "women-accessories-hats",
      "men-clothing-suits",
      "men-clothing-pants",
      "women-clothing-jackets",
      "women-clothing-tops",
      "men-clothing-shirts",
      "men-clothing-sport",
      "women-clothing-hoodies"
    ],
    colors: [
      "black",
      "blue",
      "white",
      "light blue",
      "rose",
      "gray",
      "navy blue",
      "dark gray",
      "wine",
      "navy",
      "orange",
      "burgundy",
      "light brown",
      "blue black",
      "beige",
      "green",
      "dark beige",
      "red",
      "yellow"
    ],
    _id: "5c70203d29b47b30640a8f72",
    value: "s",
    sizeDesc: "small",
    __v: 0
  },
  {
    prices: {
      min: 55,
      max: 655.55
    },
    categories: [
      "women-clothing-dresses",
      "men-clothing-suits",
      "women-clothing-jackets",
      "women-clothing-pants",
      "men-clothing-pants",
      "men-clothing-shirts",
      "men-clothing-sport",
      "women-accessories-hats"
    ],
    colors: [
      "dark beige",
      "blue black",
      "gray",
      "black",
      "white",
      "navy",
      "orange",
      "dark gray",
      "navy blue",
      "blue",
      "green",
      "red",
      "rose",
      "yellow",
      "burgundy",
      "beige"
    ],
    _id: "5c70209d29b47b30640a8f75",
    value: "xl",
    sizeDesc: "extra large",
    __v: 0
  }
];

describe("AdmSizes component", () => {
  it("AdmSizes component render", () => {
    const wrapper = shallow(
      <AdmSizes
        isExistingSizesFetching={isExistingSizesFetching}
        existingSizes={existingSizes}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
