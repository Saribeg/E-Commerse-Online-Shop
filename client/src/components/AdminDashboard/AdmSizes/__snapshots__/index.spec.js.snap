// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdmSizes component AdmSizes component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(AdmSizes)
    existingSizes={
      Array [
        Object {
          "__v": 0,
          "_id": "5c73fd305adcc004e054d124",
          "categories": Array [
            "women-shoes",
          ],
          "colors": Array [
            "black",
            "gray",
            "blue",
            "rose",
            "dark beige",
            "white",
            "red",
            "navy",
          ],
          "prices": Object {
            "max": 209.99,
            "min": 63,
          },
          "value": "36",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd405adcc004e054d128",
          "categories": Array [
            "men-shoes",
            "women-shoes",
          ],
          "colors": Array [
            "black",
            "brown",
            "dark beige",
            "rose",
          ],
          "prices": Object {
            "max": 209.99,
            "min": 88.3,
          },
          "value": "40",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd455adcc004e054d129",
          "categories": Array [
            "men-shoes",
          ],
          "colors": Array [
            "black",
          ],
          "prices": Object {
            "max": 122.32,
            "min": 122.32,
          },
          "value": "41",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd495adcc004e054d12a",
          "categories": Array [
            "men-shoes",
          ],
          "colors": Array [
            "black",
          ],
          "prices": Object {
            "max": 122.32,
            "min": 122.32,
          },
          "value": "42",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd4c5adcc004e054d12b",
          "categories": Array [
            "men-shoes",
          ],
          "colors": Array [
            "black",
          ],
          "prices": Object {
            "max": 122.32,
            "min": 122.32,
          },
          "value": "43",
        },
        Object {
          "__v": 0,
          "_id": "5c7d1efa558cad33d8f43f4e",
          "categories": Array [
            "men-accessories-belts",
          ],
          "colors": Array [
            "gray",
            "burgundy",
          ],
          "prices": Object {
            "max": 55,
            "min": 55,
          },
          "value": "76 - 81 CM",
        },
        Object {
          "__v": 0,
          "_id": "5c7d1f0f558cad33d8f43f4f",
          "categories": Array [
            "men-accessories-belts",
          ],
          "colors": Array [
            "gray",
            "burgundy",
            "brown",
            "navy blue",
          ],
          "prices": Object {
            "max": 55,
            "min": 55,
          },
          "value": "81 - 86 CM",
        },
        Object {
          "__v": 0,
          "_id": "5c7d1f17558cad33d8f43f50",
          "categories": Array [
            "men-accessories-belts",
          ],
          "colors": Array [
            "gray",
            "burgundy",
            "brown",
            "navy blue",
          ],
          "prices": Object {
            "max": 55,
            "min": 55,
          },
          "value": "86 - 91 CM",
        },
        Object {
          "__v": 0,
          "_id": "5c70205129b47b30640a8f73",
          "categories": Array [
            "women-clothing-dresses",
            "women-accessories-hats",
            "men-clothing-suits",
            "men-clothing-pants",
            "women-clothing-jackets",
            "women-clothing-pants",
            "women-clothing-tops",
            "men-clothing-shirts",
            "men-clothing-sport",
            "women-clothing-hoodies",
          ],
          "colors": Array [
            "black",
            "white",
            "dark beige",
            "blue",
            "yellow",
            "blue black",
            "gray",
            "dark gray",
            "navy blue",
            "navy",
            "burgundy",
            "light brown",
            "beige",
            "green",
            "red",
            "rose",
            "light blue",
            "orange",
            "wine",
          ],
          "prices": Object {
            "max": 655.55,
            "min": 20,
          },
          "sizeDesc": "medium",
          "value": "m",
        },
        Object {
          "__v": 0,
          "_id": "5c700f151c7b2913f88b834c",
          "categories": Array [
            "women-clothing-dresses",
            "women-clothing-pants",
            "men-clothing-suits",
            "men-clothing-pants",
            "women-clothing-jackets",
            "women-clothing-tops",
            "women-accessories-hats",
            "men-clothing-shirts",
            "men-clothing-sport",
            "women-clothing-hoodies",
          ],
          "colors": Array [
            "black",
            "blue",
            "light blue",
            "rose",
            "yellow",
            "gray",
            "navy blue",
            "dark gray",
            "wine",
            "white",
            "navy",
            "dark beige",
            "burgundy",
            "light brown",
            "blue black",
            "beige",
            "green",
            "red",
            "orange",
          ],
          "prices": Object {
            "max": 575,
            "min": 30,
          },
          "sizeDesc": "extra small",
          "value": "xs",
        },
        Object {
          "__v": 0,
          "_id": "5c7d1f1f558cad33d8f43f51",
          "categories": Array [
            "men-accessories-belts",
            "women-accessories-belts",
          ],
          "colors": Array [
            "burgundy",
            "brown",
            "dark beige",
            "black",
            "red",
            "beige",
            "white",
            "rose",
          ],
          "prices": Object {
            "max": 159.95,
            "min": 25.95,
          },
          "value": "91 - 96 CM",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd345adcc004e054d125",
          "categories": Array [
            "women-shoes",
          ],
          "colors": Array [
            "black",
            "gray",
            "dark beige",
            "blue",
            "rose",
            "wine",
            "white",
            "red",
            "navy",
            "navy blue",
          ],
          "prices": Object {
            "max": 209.99,
            "min": 63,
          },
          "value": "37",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd385adcc004e054d126",
          "categories": Array [
            "women-shoes",
            "men-shoes",
          ],
          "colors": Array [
            "black",
            "gray",
            "brown",
            "dark beige",
            "wine",
            "white",
            "red",
            "navy",
            "navy blue",
          ],
          "prices": Object {
            "max": 209.99,
            "min": 63,
          },
          "value": "38",
        },
        Object {
          "__v": 0,
          "_id": "5c73fd3c5adcc004e054d127",
          "categories": Array [
            "women-shoes",
            "men-shoes",
          ],
          "colors": Array [
            "gray",
            "black",
            "brown",
            "dark beige",
            "blue",
            "rose",
            "wine",
            "white",
            "red",
            "navy",
            "navy blue",
          ],
          "prices": Object {
            "max": 209.99,
            "min": 63,
          },
          "value": "39",
        },
        Object {
          "__v": 0,
          "_id": "5c9923bf4f16a51810eb119f",
          "categories": Array [
            "men-accessories-clocks",
          ],
          "colors": Array [
            "brown",
            "black",
          ],
          "prices": Object {
            "max": 450,
            "min": 399.99,
          },
          "value": "Standard",
        },
        Object {
          "__v": 0,
          "_id": "5c9923db4f16a51810eb11a0",
          "categories": Array [
            "men-accessories-clocks",
          ],
          "colors": Array [
            "brown",
            "black",
          ],
          "prices": Object {
            "max": 450,
            "min": 400,
          },
          "value": "Small",
        },
        Object {
          "__v": 0,
          "_id": "5c9923e34f16a51810eb11a1",
          "categories": Array [
            "men-accessories-clocks",
          ],
          "colors": Array [
            "brown",
          ],
          "prices": Object {
            "max": 450,
            "min": 399.99,
          },
          "value": "Midsize",
        },
        Object {
          "__v": 0,
          "_id": "5c9923ec4f16a51810eb11a2",
          "categories": Array [
            "men-accessories-clocks",
          ],
          "colors": Array [
            "brown",
            "black",
          ],
          "prices": Object {
            "max": 450,
            "min": 400,
          },
          "value": "XL/Oversize",
        },
        Object {
          "__v": 0,
          "_id": "5c9924354f16a51810eb11a3",
          "categories": Array [
            "women-accessories-wallets",
            "men-accessories-wallets",
            "kids-girls-toys",
          ],
          "colors": Array [
            "white",
            "black",
            "brown",
            "burgundy",
            "blue",
            "yellow",
            "orange",
            "green",
            "beige",
            "wine",
            "rose",
            "dark beige",
            "red",
            "navy blue",
            "dark gray",
            "violet",
          ],
          "prices": Object {
            "max": 70,
            "min": 23,
          },
          "value": "standart",
        },
        Object {
          "__v": 0,
          "_id": "5c70206129b47b30640a8f74",
          "categories": Array [
            "women-clothing-dresses",
            "women-clothing-pants",
            "men-clothing-suits",
            "men-clothing-pants",
            "women-clothing-jackets",
            "women-clothing-tops",
            "women-accessories-hats",
            "men-clothing-shirts",
            "men-clothing-sport",
            "women-clothing-hoodies",
          ],
          "colors": Array [
            "black",
            "blue",
            "light blue",
            "dark beige",
            "rose",
            "blue black",
            "gray",
            "navy blue",
            "dark gray",
            "wine",
            "white",
            "navy",
            "orange",
            "light brown",
            "beige",
            "green",
            "red",
            "burgundy",
            "yellow",
          ],
          "prices": Object {
            "max": 655.55,
            "min": 20,
          },
          "sizeDesc": "large",
          "value": "l",
        },
        Object {
          "__v": 0,
          "_id": "5c70203d29b47b30640a8f72",
          "categories": Array [
            "women-clothing-dresses",
            "women-clothing-pants",
            "women-accessories-hats",
            "men-clothing-suits",
            "men-clothing-pants",
            "women-clothing-jackets",
            "women-clothing-tops",
            "men-clothing-shirts",
            "men-clothing-sport",
            "women-clothing-hoodies",
          ],
          "colors": Array [
            "black",
            "blue",
            "white",
            "light blue",
            "rose",
            "gray",
            "navy blue",
            "dark gray",
            "wine",
            "navy",
            "orange",
            "burgundy",
            "light brown",
            "blue black",
            "beige",
            "green",
            "dark beige",
            "red",
            "yellow",
          ],
          "prices": Object {
            "max": 655.55,
            "min": 20,
          },
          "sizeDesc": "small",
          "value": "s",
        },
        Object {
          "__v": 0,
          "_id": "5c70209d29b47b30640a8f75",
          "categories": Array [
            "women-clothing-dresses",
            "men-clothing-suits",
            "women-clothing-jackets",
            "women-clothing-pants",
            "men-clothing-pants",
            "men-clothing-shirts",
            "men-clothing-sport",
            "women-accessories-hats",
          ],
          "colors": Array [
            "dark beige",
            "blue black",
            "gray",
            "black",
            "white",
            "navy",
            "orange",
            "dark gray",
            "navy blue",
            "blue",
            "green",
            "red",
            "rose",
            "yellow",
            "burgundy",
            "beige",
          ],
          "prices": Object {
            "max": 655.55,
            "min": 55,
          },
          "sizeDesc": "extra large",
          "value": "xl",
        },
      ]
    }
    isExistingSizesFetching={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
